tracking/descarga_bsp_esios.py

```
@@ -0,0 +1,194 @@
+            if key_pem:
+                with tempfile.NamedTemporaryFile(delete=False, suffix=".pem") as key_file:
+                    key_file.write(key_pem.encode("utf-8"))
+                    key_path = key_file.name
+                    self._temp_files.append(key_path)  # Track for cleanup
+
+            self.client_cert = [cert_path, key_path] #set the client certificate and key paths as class attributes 
+
+            return cert_pem, key_pem
+
+        except ValueError as e:
+            raise ValueError(f"Failed to load .p12 file: {e}. Check if the password is correct or missing.")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)